### PR TITLE
fix(expo-cli): add warning about damaged simulator builds catalina

### LINKED ISSUE #1197: iOS Simulator Builds damaged error on MacOS Catalina
## Description

The most recent iOS simulator builds that I have created are all damaged. I am able to launch the app on the simulator, but it fails to proceed past the splash screen.

## Expected Behavior

After running `xcrun simctl install booted <app path>`, I should be able to open the app and proceed past the splash screen.

## Observed Behavior

After running `xcrun simctl install booted <app path>`, I am stuck at the splash screen after launching the app. A system warning prompt is displayed to me saying that the app is damaged.

Screen Shot 2019-11-10 at 10 40 03 PM

Screen Shot 2019-11-10 at 10 52 59 PM

## Environment
```
Expo CLI 3.4.1 environment info:
    System:
      OS: macOS 10.15.1
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 12.9.1 - ~/.nvm/versions/node/v12.9.1/bin/node
      Yarn: 1.12.3 - /usr/local/bin/yarn
      npm: 6.12.1 - ~/.nvm/versions/node/v12.9.1/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.4 AI-183.6156.11.34.5692245
      Xcode: 11.2/11B52 - /usr/bin/xcodebuild
    npmPackages:
      @types/react: 16.8.3 => 16.8.3
      @types/react-native: 0.60.13 => 0.60.13
      @types/react-navigation: ^1.5.15 => 1.5.15
      expo: ^34.0.1 => 34.0.4
      react: 16.8.3 => 16.8.3
      react-native: https://github.com/expo/react-native/archive/sdk-34.0.0.tar.gz => 0.59.8
      react-navigation: ^1.0.0-beta.21 => 1.6.1
    npmGlobalPackages:
      expo-cli: 3.4.1
```
Also specify:

- Phone/emulator/simulator platform, model and version: `Simulator - iPhone 11 Pro Max - 13.2`

## Reproducible Demo

1. **Download** zipped app bundle from here: https://exp-shell-app-assets.s3.us-west-1.amazonaws.com/ios/%40foodbarrio/foodbarrio-qa-3175d259-0256-49c0-a707-775191c1b9da-simulator.tar.gz
2. **Unzip** foodbarrio-qa-3175d259-0256-49c0-a707-775191c1b9da-simulator.tar.gz
3. **Launch** one simulator
4. **Run** `xcrun simctl install booted foodbarrio-qa.app
5. **Open** the `foodbarrioQA` app on the simulator
6. **Encounter** the observed behavior as mentioned above in the **Observed Behavior** section

### PR DESCRIPTION
**Pull request by byCedric**
Fixes #1197

It's just a warning, styled like the `expo upgrade` warnings, to tell the developer about possible damaged simulator builds. I put it in twice, so it's always visible when the user starts the build and when the build is completed (or sent, with `--no-wait`).

<details>
<summary>This is how it looks like when starting</summary>

</details>

<details>
<summary>This is how it looks like at the end</summary>

</details>

### CODE REVIEW

File: packages/expo-cli/src/commands/build/ios/IOSBuilder.ts [hunk 1]
```diff
@@ -41,6 +41,8 @@ class IOSBuilder extends BaseBuilder {
 
   async run(): Promise<void> {
     await this.validateProject();
+    this.maybeWarnDamagedSimulator();
```
Comment (byCedric) [inline packages/expo-cli/src/commands/build/ios/IOSBuilder.ts:44]: We can push this below the build in progress check too. The reason here is to display it ASAP when it makes sense (after validation).

Comment (brentvatne) [reply packages/expo-cli/src/commands/build/ios/IOSBuilder.ts:44]: can we check operating system version to only show this on catalina+?

Comment (byCedric) [reply packages/expo-cli/src/commands/build/ios/IOSBuilder.ts:46]: Done 😄

Comment (brentvatne) [reply packages/expo-cli/src/commands/build/ios/IOSBuilder.ts:46]: can you also create an entry on expo.fyi that explains this and maybe provides some context on why it happens?

File: packages/expo-cli/src/commands/build/ios/IOSBuilder.ts [hunk 2]
```diff
@@ -322,6 +327,26 @@ See https://docs.expo.io/versions/latest/distribution/building-standalone-apps/#
...
+    if (isMacOsCatalinaOrLater && this.options.type === 'simulator') {
+      log.newLine();
+      log(
+        chalk.bold(
+          `🚨 If the build is not installable on your simulator because of "${chalk.underline(
+            `... is damaged and can't be opened.`
+          )}", please run:`
+        )
+      );
+      log(chalk.grey.bold('xattr -rd com.apple.quarantine /path/to/your.app'));
```
Comment (brentvatne) [inline packages/expo-cli/src/commands/build/ios/IOSBuilder.ts:347]: we should provide a full set of instructions on how to run this, perhaps on the expo.fyi page

eg: we should tell people they need to extract the tar.gz and then drag into simulator

**Pull request merged**